docs/lang/articles/contribution/development_tips.md

```diff
@@ -0,0 +1,102 @@
+---
+sidebar_position: 3
+---
+
+# Development tips
+
+This section gives you some tips on the Taichi compiler development.
+Please make sure you have gone through [developer installation](./dev_install).
+
+## Workflow of the Taichi compiler
+
+[Life of a Taichi kernel](./compilation.md) may be worth checking out. It
+explains the whole compilation process.
+
+## Folder structure
+
+Key folders: _(the following chart can be generated by [`tree . -L 2`](https://linux.die.net/man/1/tree))_
+
+```
+.
+├── benchmarks              # Performance benchmarks
+├── docs                    # Documentation
+├── examples                # Examples
+├── external                # External libraries
+├── misc                    # Random yet useful files
+├── python                  # Python frontend implementation
+│   ├── core                # Loading & interacting with Taichi core
+│   ├── lang                # Python-embbed Taichi language & syntax (major)
+│   ├── snode               # Structure nodes
+│   ├── tools               # Handy end-user tools
+│   └── misc                # Miscellaneous utilities
+├── taichi                  # The core compiler implementation
```

Comment:
It might be worthing noting that taichi/ folder is C++ files of libtaichi_core.so.

Maybe sth like: The core compiler implementation: C++ source of libtaichi_core.so.